packages/shared_preferences/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.java

```diff
@@ -116,30 +116,44 @@ private String encodeList(List<String> list) throws IOException {
     return filteredPrefs;
   }
 
+  void backgroundTask(final Editor editor, final MethodChannel.Result result) {
+    new Thread(
+            new Runnable() {
+              @Override
+              public void run() {
+                boolean status = editor.commit();
+                result.success(status);
```

Comment:
You can't do this. result objects need to be called from UiThread. You can use `AsyncTask` if you want a nice API to work with.